symfony/client/src/orangehrmPimPlugin/pages/employee/EditCustomField.vue

```diff
@@ -0,0 +1,197 @@
+
+      <oxd-divider />
+
+      <oxd-form :loading="isLoading" @submitValid="onSave">
+        <oxd-form-row>
+          <oxd-grid :cols="2" class="orangehrm-full-width-grid">
+            <oxd-grid-item class="organization-name-container">
+              <oxd-input-field
+                label="Field Name"
+                v-model="customField.fieldName"
+                :rules="rules.fieldName"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                label="Screen"
+                type="dropdown"
+                v-model="customField.screen"
+                :rules="rules.screen"
+                :options="screenList"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :cols="2" class="orangehrm-full-width-grid">
+            <oxd-grid-item class="organization-name-container">
+              <oxd-input-field
+                label="Type"
+                type="dropdown"
+                v-model="customField.fieldType"
+                :rules="rules.fieldType"
+                :options="fieldTypeList"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item v-if="isDropDownField">
+              <oxd-input-field
+                label="Select Options"
+                v-model="customField.extraData"
+                :rules="rules.extraData"
+                :required="isDropDownField"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            type="button"
+            displayType="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@orangehrm/core/util/services/api.service';
+import { required, shouldNotExceedCharLength } from '@/core/util/validation/rules'
+
+const customFieldModel = {
+  fieldName: '',
+  screen: '',
+  fieldType: '',
+  extraData: '',
+};
+
+export default {
+  props: {
+    customFieldId: {
+      type: Number,
+      required: true,
+    },
+    screenList: {
+      type: Array,
+      required: true,
+    },
+    fieldTypeList: {
+      type: Array,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      customField: {...customFieldModel},
+      rules: {
+        fieldName: [required, shouldNotExceedCharLength(250)],
+        screen: [shouldNotExceedCharLength(100)],
+        fieldType: [required, shouldNotExceedCharLength(15)],
+        extraData: [shouldNotExceedCharLength(250)],
```

Comment:
Select options should be a required field if field type selected is dropdown. please add appropriate validations.

Reply:
added to the isDropDownField() function since it is dynamically changing.